src/libYARP_math/src/math.cpp

```diff
@@ -953,3 +953,67 @@ Matrix yarp::math::adjointInv(const Matrix &H)
 
     return A;
 }
+
+
+PointCloud<DataXYZ> yarp::math::depthToPC(const yarp::sig::ImageOf<PixelFloat> &depth,
+                                          const yarp::sig::IntrinsicParams &intrinsic)
+{
+    yAssert(depth.width()  != 0);
+    yAssert(depth.height() != 0);
+    size_t w = depth.width();
+    size_t h = depth.height();
+    PointCloud<DataXYZ> pointCloud;
+    pointCloud.resize(w, h);
+
+    for (size_t u = 0; u < w; ++u) {
+        for (size_t v = 0; v < h; ++v) {
+            pointCloud(u,v).x = (u - intrinsic.principalPointX)/intrinsic.focalLengthX*depth.pixel(u,v);
+            pointCloud(u,v).y = (v - intrinsic.principalPointY)/intrinsic.focalLengthY*depth.pixel(u,v);
+            pointCloud(u,v).z = depth.pixel(u,v);
+        }
+    }
+    return pointCloud;
+}
+
+template<typename T1, typename T2>
+yarp::sig::PointCloud<T1> yarp::math::depthRgbToPC(const yarp::sig::ImageOf<yarp::sig::PixelFloat>& depth,
+                                                   const yarp::sig::ImageOf<T2>& color,
+                                                   const yarp::sig::IntrinsicParams& intrinsic)
+{
+    yAssert(depth.width()  != 0);
+    yAssert(depth.height() != 0);
+    yAssert(depth.width()  == color.width());
+    yAssert(depth.height() == color.height());
+    size_t w = depth.width();
+    size_t h = depth.height();
+    PointCloud<T1> pointCloud;
+    pointCloud.resize(w, h);
+
+    for (size_t u = 0; u < w; ++u) {
+        for (size_t v = 0; v < h; ++v) {
+            // Depth
+            pointCloud(u,v).x = (u - intrinsic.principalPointX)/intrinsic.focalLengthX*depth.pixel(u,v);
```

Comment:
As for the other parameters, it would be good if we could reference a source for this equation.